### PR TITLE
[New Rule] PowerShell Keylogging Script

### PR DESCRIPTION
**Pull request by w0rk3r**
## Summary

Detects the use of Win32 API Functions that can be used to capture user Keystrokes in Powershell Scripts.
Attackers use this technique to capture user input, looking for credentials and/or other valuable data.

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/windows/collection_posh_keylogger.toml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+type = "query"
+
+query = '''
+event.code:"4104" and 
+  powershell.file.script_block_text : (
+    GetAsyncKeyState or
+    GetKeyboardState or
+    SetWindowsHookEx or
+    SetWindowsHookExA
+  )
```
Comment (Samirbous) [inline rules/windows/collection_posh_keylogger.toml:35]:
```suggestion
event.code:"4104" and 
  ( 
   powershell.file.script_block_text : (GetAsyncKeyState or NtUserGetAsyncKeyState GetKeyboardState or Get-Keystrokes) or 
   powershell.file.script_block_text : ((SetWindowsHookA or SetWindowsHookW or SetWindowsHookEx or SetWindowsHookExA) and (GetForegroundWindow or GetWindowTextA or GetWindowTextW or WM_KEYBOARD_LL))
   )
```

File: rules/windows/collection_posh_keylogger.toml [hunk 2]
```diff
@@ -0,0 +1,75 @@
+[metadata]
+creation_date = "2021/10/15"
+maturity = "production"
+updated_date = "2021/10/15"
+
+[rule]
+author = ["Elastic"]
+description = """
+This rule detects the use of Win32 API Functions that can be used to capture user Keystrokes in Powershell Scripts.
+Attackers use this technique to capture user input, looking for credentials and/or other valuable data.,
```
Comment (Samirbous) [inline rules/windows/collection_posh_keylogger.toml:10]:
```suggestion
Attackers use this technique to capture user input, looking for credentials and/or other valuable data.
```

File: rules/windows/collection_posh_keylogger.toml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+]
+risk_score = 73
+rule_id = "bd2c86a0-8b61-4457-ab38-96943984e889"
+severity = "high"
+tags = ["Elastic", "Host", "Windows", "Threat Detection", "Collection"]
+timestamp_override = "event.ingested"
+type = "query"
+
+query = '''
+event.code:"4104" and
```
Comment (brokensound77) [inline rules/windows/collection_posh_keylogger.toml:29]:
```suggestion
event.category:process and 
```

File: rules/windows/collection_posh_keylogger.toml [hunk 4]
```diff
@@ -0,0 +1,73 @@
+[metadata]
+creation_date = "2021/10/15"
+maturity = "production"
+updated_date = "2021/10/15"
+
+[rule]
+author = ["Elastic"]
+description = """
+This rule detects the use of Win32 API Functions that can be used to capture user Keystrokes in Powershell Scripts.
+Attackers use this technique to capture user input, looking for credentials and/or other valuable data.
```
Comment (brokensound77) [inline rules/windows/collection_posh_keylogger.toml:10]:
```suggestion
Detects the use of Win32 API Functions that can be used to capture user Keystrokes in PowerShell Scripts.
Attackers use this technique to capture user input, looking for credentials and/or other valuable data.
```

**Review by brokensound77 — APPROVED** [review]
Left a few comments then LGTM 👍

**Pull request merged**